build/source/engine/getVectorz.f90

```diff
@@ -0,0 +1,371 @@
+                        sMul,                    & ! intent(out):   multiplier for state vector (used in the residual calculations)
+                        dMat,                    & ! intent(out):   diagonal of the Jacobian matrix (excludes fluxes) 
+                        err,message)               ! intent(out):   error control
+ ! --------------------------------------------------------------------------------------------------------------------------------
+ ! --------------------------------------------------------------------------------------------------------------------------------
+ ! input
+ logical(lgt),intent(in)         :: computeVegFlux         ! flag to indicate if we are computing fluxes over vegetation (.false. means veg is buried with snow)
+ real(dp),intent(in)             :: canopyDepth            ! canopy depth (m)
+ type(var_dlength),intent(in)    :: prog_data              ! prognostic variables for a local HRU
+ type(var_dlength),intent(in)    :: diag_data              ! diagnostic variables for a local HRU
+ type(var_ilength),intent(in)    :: indx_data              ! indices defining model states and layers
+ ! output: state vectors
+ real(dp),intent(out)            :: stateVec(:)            ! model state vector (mixed units)
+ real(dp),intent(out)            :: fScale(:)              ! function scaling vector (mixed units)
+ real(dp),intent(out)            :: xScale(:)              ! variable scaling vector (mixed units)
+ real(qp),intent(out)            :: sMul(:)    ! NOTE: qp  ! multiplier for state vector (used in the residual calculations)
+ real(dp),intent(out)            :: dMat(:)                ! diagonal of the Jacobian matrix (excludes fluxes)
+ ! output: error control
+ integer(i4b),intent(out)        :: err                    ! error code
+ character(*),intent(out)        :: message                ! error message
+ ! --------------------------------------------------------------------------------------------------------------------------------
+ ! local variables
+ ! --------------------------------------------------------------------------------------------------------------------------------
+ ! scaling parameters
+ real(dp),parameter              :: fScaleLiq=0.01_dp      ! func eval: characteristic scale for volumetric liquid water content (-)
+ real(dp),parameter              :: fScaleMat=10._dp       ! func eval: characteristic scale for matric head (m)
+ real(dp),parameter              :: fScaleNrg=1000000._dp  ! func eval: characteristic scale for energy (J m-3)
+ real(dp),parameter              :: xScaleLiq=0.1_dp       ! state var: characteristic scale for volumetric liquid water content (-)
+ real(dp),parameter              :: xScaleMat=10._dp       ! state var: characteristic scale for matric head (m)
+ real(dp),parameter              :: xScaleTemp=1._dp       ! state var: characteristic scale for temperature (K)
```

Comment:
Add brief comment how scaling parameters are selected (or refer to appropriate documentation)